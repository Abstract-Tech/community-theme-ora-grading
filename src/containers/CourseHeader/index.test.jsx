import React from 'react';
import { shallow } from 'enzyme';

import { AppContext } from '@edx/frontend-platform/react';
import { Header } from '.';

jest.mock('./AnonymousUserMenu', () => 'AnonymousUserMenu');
jest.mock('./AuthenticatedUserDropdown', () => 'AuthenticatedUserDropdown');
jest.mock('./LinkedLogo', () => 'LinkedLogo');
jest.mock('./CourseLabel', () => 'CourseLabel');

jest.mock('@edx/frontend-platform/react', () => ({
  AppContext: { authenticatedUser: null },
}));
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: (context) => context,
}));

const courseData = {
  courseOrg: 'course-org',
  courseNumber: 'course-number',
  courseTitle: 'course-title',
};

describe('Header component', () => {
  const props = {
    ...courseData,
    intl: { formatMessage: (msg) => msg.defaultMessage },
  };
  test('snapshot', () => {
    expect(shallow(<Header {...props} />)).toMatchSnapshot();
  });
  test('snapshot with authenticatedUser', () => {
    AppContext.authenticatedUser = { username: 'test' };
    expect(shallow(<Header {...props} />)).toMatchSnapshot();
  });
});
