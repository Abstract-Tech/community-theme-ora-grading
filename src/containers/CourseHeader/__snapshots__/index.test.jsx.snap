// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component snapshot 1`] = `
<header
  className="course-header"
>
  <a
    className="sr-only sr-only-focusable"
    href="#main-content"
  >
    Skip to main content.
  </a>
  <div
    className="container-xl py-2 d-flex align-items-center"
  >
    <LinkedLogo />
    <CourseLabel
      courseNumber="course-number"
      courseOrg="course-org"
      courseTitle="course-title"
    />
    <AnonymousUserMenu />
  </div>
</header>
`;

exports[`Header component snapshot with authenticatedUser 1`] = `
<header
  className="course-header"
>
  <a
    className="sr-only sr-only-focusable"
    href="#main-content"
  >
    Skip to main content.
  </a>
  <div
    className="container-xl py-2 d-flex align-items-center"
  >
    <LinkedLogo />
    <CourseLabel
      courseNumber="course-number"
      courseOrg="course-org"
      courseTitle="course-title"
    />
    <AuthenticatedUserDropdown
      username="test"
    />
  </div>
</header>
`;
